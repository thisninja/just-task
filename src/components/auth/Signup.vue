<template>
  <div class="full-width">
    <img
      class="layout__icon"
      src="@/assets/task-icon.jpg"
      />
    <h3>
      {{ CREATE_ACCOUNT_TITLE }}
    </h3>
  </div>
</template>

<script>
import {
  CREATE_ACCOUNT_TITLE,
} from './constants';

export default {
  name: 'Signup',
  data () {
    return {
      CREATE_ACCOUNT_TITLE,
    }
  },
}
</script>
