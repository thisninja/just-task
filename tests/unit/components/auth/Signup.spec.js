import Vuex from 'vuex';
import VueRouter from 'vue-router';
import VueMaterial from 'vue-material'
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Signup from '@/components/auth/Signup.vue';

const localVue = createLocalVue();

localVue.use(VueMaterial);
localVue.use(VueRouter);
localVue.use(Vuex);

describe('Signup.vue', () => {
  let wrapper;
  let store;
  let actions;
  let mutations;
  let getters;

  beforeEach(() => {
    getters = getters || {
      isAuthenticated: () => true,
    };

    actions = actions || {
      signup: jest.fn(),
    };

    mutations = mutations || {
      authUser: jest.fn(),
    };

    store = new Vuex.Store({
      getters,
      actions,
      mutations,
    });

    wrapper = shallowMount(Signup, {
      store,
      localVue,
    });
  });

  describe('snapshots', () => {
    it('should match snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    describe('onSubmit', () => {
      afterEach(() => {
        jest.clearAllMocks();
      });

      it('should trigger onSubmit', () => {
        const mock = jest.fn();
        wrapper.vm.onSubmit = mock;
        wrapper.find(".auth-form").trigger("submit.prevent")

        expect(mock).toHaveBeenCalled();
      });

      it('should trigger isValidFormByType with param "signup"', () => {
        const mock = jest.fn();
        wrapper.vm.isValidFormByType = mock;
        wrapper.vm.onSubmit();

        expect(mock).toHaveBeenCalledWith('signup');
        expect(actions.signup).not.toHaveBeenCalled();
      });

      it('should trigger signup action if isValidFormByType returned true', () => {
        const mock = jest.fn().mockImplementationOnce(() => true);
        wrapper.vm.isValidFormByType = mock;
        wrapper.vm.onSubmit();

        expect(actions.signup).toHaveBeenCalled();
      });
    });
  });
});
