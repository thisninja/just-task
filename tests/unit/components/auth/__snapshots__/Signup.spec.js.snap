// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup.vue snapshots should match snapshot 1`] = `
<div class="full-width">
  <common-placeholder-stub title="Create an accout to use the app"></common-placeholder-stub>
  <!---->
  <form novalidate="novalidate" class="auth-form">
    <md-field-stub mdcounter="true" mdtogglepassword="true" class=""><label>
        Enter your email
      </label>
      <md-input-stub value="" id="md-input-8no90gn6" type="text"></md-input-stub>
    </md-field-stub>
    <md-field-stub mdcounter="true" mdtogglepassword="true" class=""><label>
        Enter your password
      </label>
      <md-input-stub value="" id="md-input-khx7e1g16" type="password"></md-input-stub>
    </md-field-stub>
    <md-field-stub mdcounter="true" mdtogglepassword="true" class=""><label>
        Confirm your password
      </label>
      <md-input-stub value="" id="md-input-n9zd3mtag" type="password"></md-input-stub>
    </md-field-stub>
    <md-button-stub mdripple="true" type="submit" class="md-raised md-primary">
      Sign up
    </md-button-stub>
  </form>
  <router-link-stub to="/login" tag="a" event="click" class="layout__link">
    Already have an account?&nbsp;
    <span>
      Login
    </span></router-link-stub>
</div>
`;
